RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,301 @@
+
+The majority of all communication on the Tari communication network will be performed using direct P2P channels established between different CCs and CNs once they are aware of the peer addresses of each other that contain their online communication addresses.
+Message propagation on the network will typically consist only of joining and discovery requests where a CC or CN wants to join the network or retrieve the peer address of another CC or CN so that a direct P2P channel can be established.
+
+Messages can be transmitted in this network in either an unencrypted or encrypted form.
+Typically messages that have been sent in unencrypted form is of interest to a number of CNs on the network and should be propagated so that every CN that is interested in that data message obtains a copy.
+Block and Transaction propagation are examples of data messages where multiple entities on the Tari communication network are interested in that data message, this requires propagation through the entire Tari communication network in unencrypted form.
+
+Encrypted data messages make use of the source and destinations identification cryptographic keys to construct a shared secret with which the message can be encoded and decoded.
+This ensures that only the two parties are able to decode the data message as it is propagated through the communication network.
+This mechanism can be used to perform private discovery requests, where the online communication address of the source node is encrypted and propagated through the network until it reached the destination node.
+Private discovery requests can only be performed if both parties are online at the same time.
+Encryption of the data message ensuring that only the destination node is able to view the online address of the source node as the data message moves through the network.
+Once the destination node receives and decrypts the data message, that node is then able to establish a P2P communication channel with the source node for any further communication.
+
+Propagation of completely private discovery request, hidden as an encrypted data message, can be performed as a broadcast through the entire network using the Gossip protocol.
+Propagation of public discovery requests can be performed using more efficient directed propagation using the Kademlia protocol.
+As encrypted message with visible destinations tend to not be of interest to the rest of the network, directed propagation using the Kademlia protocol to forward these messages to the correct parties are preferred.
+
+This same encryption technique can be used to send encrypted messages to a single node or a group of nodes, where the group of nodes have shared identification keys.
+A Validation Committee is an example of a group of CNs that have shared identification keys for the committee.
+The shared identification keys ensure that all members of that committee are able to receive and decrypt data messages that were sent to the committee.
```

Comment:
A 1-of-n signature could work, but my fear is that it will become a single common point making it 1-of-1. I think this is true for the Shamir's secret-sharing scheme but maybe not for other 1-of-n signature schemes. I believe the Asset Creator should create the shared identification keys (for the committee) during the Asset Creation process.